{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"provenance":[]},"kernelspec":{"name":"python3","display_name":"Python 3"},"language_info":{"name":"python"}},"cells":[{"cell_type":"code","source":["!pip install --upgrade -q landsatxplore"],"metadata":{"id":"aWohXRHsVGfE"},"execution_count":null,"outputs":[]},{"cell_type":"code","source":["pip install fastapi"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"66dpQcfgSbs6","executionInfo":{"status":"ok","timestamp":1707590034976,"user_tz":480,"elapsed":6753,"user":{"displayName":"Natasha Thussu Dhar","userId":"06843221064364226659"}},"outputId":"21e05ae4-1cb8-4b33-e80a-ee567a393bb3"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["Requirement already satisfied: fastapi in /usr/local/lib/python3.10/dist-packages (0.109.2)\n","Requirement already satisfied: pydantic!=1.8,!=1.8.1,!=2.0.0,!=2.0.1,!=2.1.0,<3.0.0,>=1.7.4 in /usr/local/lib/python3.10/dist-packages (from fastapi) (2.6.1)\n","Requirement already satisfied: starlette<0.37.0,>=0.36.3 in /usr/local/lib/python3.10/dist-packages (from fastapi) (0.36.3)\n","Requirement already satisfied: typing-extensions>=4.8.0 in /usr/local/lib/python3.10/dist-packages (from fastapi) (4.9.0)\n","Requirement already satisfied: annotated-types>=0.4.0 in /usr/local/lib/python3.10/dist-packages (from pydantic!=1.8,!=1.8.1,!=2.0.0,!=2.0.1,!=2.1.0,<3.0.0,>=1.7.4->fastapi) (0.6.0)\n","Requirement already satisfied: pydantic-core==2.16.2 in /usr/local/lib/python3.10/dist-packages (from pydantic!=1.8,!=1.8.1,!=2.0.0,!=2.0.1,!=2.1.0,<3.0.0,>=1.7.4->fastapi) (2.16.2)\n","Requirement already satisfied: anyio<5,>=3.4.0 in /usr/local/lib/python3.10/dist-packages (from starlette<0.37.0,>=0.36.3->fastapi) (3.7.1)\n","Requirement already satisfied: idna>=2.8 in /usr/local/lib/python3.10/dist-packages (from anyio<5,>=3.4.0->starlette<0.37.0,>=0.36.3->fastapi) (3.6)\n","Requirement already satisfied: sniffio>=1.1 in /usr/local/lib/python3.10/dist-packages (from anyio<5,>=3.4.0->starlette<0.37.0,>=0.36.3->fastapi) (1.3.0)\n","Requirement already satisfied: exceptiongroup in /usr/local/lib/python3.10/dist-packages (from anyio<5,>=3.4.0->starlette<0.37.0,>=0.36.3->fastapi) (1.2.0)\n"]}]},{"cell_type":"code","source":["pip install kaleido"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"38GioteeSvT3","executionInfo":{"status":"ok","timestamp":1707589995916,"user_tz":480,"elapsed":12942,"user":{"displayName":"Natasha Thussu Dhar","userId":"06843221064364226659"}},"outputId":"5a84ba9d-c92f-4f3f-c8ed-bd3296384d12"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["Collecting kaleido\n","  Downloading kaleido-0.2.1-py2.py3-none-manylinux1_x86_64.whl (79.9 MB)\n","\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m79.9/79.9 MB\u001b[0m \u001b[31m6.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hInstalling collected packages: kaleido\n","\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n","lida 0.0.10 requires python-multipart, which is not installed.\u001b[0m\u001b[31m\n","\u001b[0mSuccessfully installed kaleido-0.2.1\n"]}]},{"cell_type":"code","source":["pip install python-multipart"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"iH7R1oKRSwxF","executionInfo":{"status":"ok","timestamp":1707590008664,"user_tz":480,"elapsed":8427,"user":{"displayName":"Natasha Thussu Dhar","userId":"06843221064364226659"}},"outputId":"a54a99fa-af6e-441d-9b4e-bf773a1dc04a"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["Collecting python-multipart\n","  Downloading python_multipart-0.0.9-py3-none-any.whl (22 kB)\n","Installing collected packages: python-multipart\n","Successfully installed python-multipart-0.0.9\n"]}]},{"cell_type":"code","source":["pip install uvicorn"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"ft2zbhqGSyMQ","executionInfo":{"status":"ok","timestamp":1707589979309,"user_tz":480,"elapsed":6274,"user":{"displayName":"Natasha Thussu Dhar","userId":"06843221064364226659"}},"outputId":"b994852f-83f7-4b3f-e5b0-4cf034d398f1"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["Collecting uvicorn\n","  Downloading uvicorn-0.27.1-py3-none-any.whl (60 kB)\n","\u001b[?25l     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m0.0/60.8 kB\u001b[0m \u001b[31m?\u001b[0m eta \u001b[36m-:--:--\u001b[0m\r\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m60.8/60.8 kB\u001b[0m \u001b[31m2.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hRequirement already satisfied: click>=7.0 in /usr/local/lib/python3.10/dist-packages (from uvicorn) (7.1.2)\n","Collecting h11>=0.8 (from uvicorn)\n","  Downloading h11-0.14.0-py3-none-any.whl (58 kB)\n","\u001b[?25l     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m0.0/58.3 kB\u001b[0m \u001b[31m?\u001b[0m eta \u001b[36m-:--:--\u001b[0m\r\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m58.3/58.3 kB\u001b[0m \u001b[31m5.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n","\u001b[?25hRequirement already satisfied: typing-extensions>=4.0 in /usr/local/lib/python3.10/dist-packages (from uvicorn) (4.9.0)\n","Installing collected packages: h11, uvicorn\n","\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\n","lida 0.0.10 requires kaleido, which is not installed.\n","lida 0.0.10 requires python-multipart, which is not installed.\u001b[0m\u001b[31m\n","\u001b[0mSuccessfully installed h11-0.14.0 uvicorn-0.27.1\n"]}]},{"cell_type":"code","source":["from landsatxplore.api import API\n","\n","# Your USGS  credentials\n","username = 'pruhtopia'\n","password = 'Sanika1077!!'\n","\n","# Initialize a new API instance\n","api = API(username, password)\n","\n","# Perform a request\n","response = api.request(endpoint=\"dataset-catalogs\")\n","print(response)\n"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"-7igmVBOS3Kd","outputId":"61257e68-0e34-4ee3-9834-fc1602e275b9","executionInfo":{"status":"ok","timestamp":1707590040276,"user_tz":480,"elapsed":1230,"user":{"displayName":"Natasha Thussu Dhar","userId":"06843221064364226659"}}},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["{'EE': 'EarthExplorer', 'GV': 'GloVis', 'HDDS': 'HDDS Explorer'}\n"]}]},{"cell_type":"code","source":["import json\n","from landsatxplore.api import API\n","from landsatxplore.earthexplorer import EarthExplorer\n","\n","# Initialize a new API instance and get an access key\n","api = API(username, password)\n","\n","# Search for Landsat TM scenes\n","scenes = api.search(\n","    dataset='landsat_tm_c2_l1',\n","    latitude=25,\n","    longitude=-79,\n","    start_date='2010-01-01',\n","    end_date='2020-12-30',\n","    max_cloud_cover=50\n",")\n","\n","print(f\"{len(scenes)} scenes found.\")\n","\n","# Process the result\n","if scenes:\n","    first_scene = scenes[0]\n","    print(first_scene['landsat_scene_id'])\n","    print(\"Downloading the first scene...\")\n","    ee = EarthExplorer(username, password)\n","    ee.download(first_scene['landsat_scene_id'], output_dir='/content/')\n","\n","    print(\"First scene downloaded.\")\n","else:\n","    print(\"No scenes found.\")\n","\n","api.logout()\n"],"metadata":{"id":"ui7t3JHzWhFM","colab":{"base_uri":"https://localhost:8080/","height":513},"executionInfo":{"status":"error","timestamp":1707590049889,"user_tz":480,"elapsed":6903,"user":{"displayName":"Natasha Thussu Dhar","userId":"06843221064364226659"}},"outputId":"a73c739d-fbd4-457e-b78d-abbd995204bd"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["9 scenes found.\n","LT50140432011147CHM02\n","Downloading the first scene...\n","Download failed with dataset id 1 of 3. Re-trying with the next one.\n","None of the archived ids succeeded! Update necessary!\n"]},{"output_type":"error","ename":"EarthExplorerError","evalue":"","traceback":["\u001b[0;31m---------------------------------------------------------------------------\u001b[0m","\u001b[0;31mEarthExplorerError\u001b[0m                        Traceback (most recent call last)","\u001b[0;32m/usr/local/lib/python3.10/dist-packages/landsatxplore/earthexplorer.py\u001b[0m in \u001b[0;36mdownload\u001b[0;34m(self, identifier, output_dir, dataset, timeout, skip, overwrite)\u001b[0m\n\u001b[1;32m    195\u001b[0m                 )\n\u001b[0;32m--> 196\u001b[0;31m                 filename = self._download(\n\u001b[0m\u001b[1;32m    197\u001b[0m                     \u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0moutput_dir\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtimeout\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mtimeout\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mskip\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mskip\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0moverwrite\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0moverwrite\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;32m/usr/local/lib/python3.10/dist-packages/landsatxplore/earthexplorer.py\u001b[0m in \u001b[0;36m_download\u001b[0;34m(self, url, output_dir, timeout, chunk_size, skip, overwrite)\u001b[0m\n\u001b[1;32m     85\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0merror_msg\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 86\u001b[0;31m                 \u001b[0;32mraise\u001b[0m \u001b[0mEarthExplorerError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0merror_msg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     87\u001b[0m             \u001b[0mdownload_url\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjson\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"url\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;31mEarthExplorerError\u001b[0m: Download is not available","\nDuring handling of the above exception, another exception occurred:\n","\u001b[0;31mEarthExplorerError\u001b[0m                        Traceback (most recent call last)","\u001b[0;32m<ipython-input-12-b7e6411382c3>\u001b[0m in \u001b[0;36m<cell line: 21>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     24\u001b[0m     \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"Downloading the first scene...\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     25\u001b[0m     \u001b[0mee\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mEarthExplorer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0musername\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpassword\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 26\u001b[0;31m     \u001b[0mee\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdownload\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfirst_scene\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'landsat_scene_id'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0moutput_dir\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'/content/'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     27\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     28\u001b[0m     \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"First scene downloaded.\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;32m/usr/local/lib/python3.10/dist-packages/landsatxplore/earthexplorer.py\u001b[0m in \u001b[0;36mdownload\u001b[0;34m(self, identifier, output_dir, dataset, timeout, skip, overwrite)\u001b[0m\n\u001b[1;32m    203\u001b[0m                 \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    204\u001b[0m                     \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'None of the archived ids succeeded! Update necessary!'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 205\u001b[0;31m                     \u001b[0;32mraise\u001b[0m \u001b[0mEarthExplorerError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    206\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mfilename\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;31mEarthExplorerError\u001b[0m: "]}]},{"cell_type":"code","source":[],"metadata":{"id":"BT_1XOwWRg0i"},"execution_count":null,"outputs":[]}]}